packages/api-db-mongodb/src/db/memberPlan.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  ConnectionResult,
+  GetMemberPlansArgs,
+  CreateMemberPlanArgs,
+  DBMemberPlanAdapter,
+  DeleteMemberPlanArgs,
+  InputCursorType,
+  LimitType,
+  MemberPlan,
+  MemberPlanSort,
+  OptionalMemberPlan,
+  SortOrder,
+  UpdateMemberPlanArgs
+} from '@wepublish/api'
+import {Collection, Db, FilterQuery, MongoCountPreferences} from 'mongodb'
+import {CollectionName, DBMemberPlan} from './schema'
+import {MaxResultsPerPage} from './defaults'
+import {Cursor} from './cursor'
+
+export class MongoDBMemberPlanAdapter implements DBMemberPlanAdapter {
+  private memberPlans: Collection<DBMemberPlan>
+  private locale: string
+
+  constructor(db: Db, locale: string) {
+    this.memberPlans = db.collection(CollectionName.MemberPlans)
+    this.locale = locale
+  }
+
+  async createMemberPlan({input}: CreateMemberPlanArgs): Promise<MemberPlan> {
+    const {ops} = await this.memberPlans.insertOne({
+      createdAt: new Date(),
+      modifiedAt: new Date(),
+      name: input.name,
+      imageID: input.imageID,
+      description: input.description,
+      isActive: input.isActive,
+      pricePerMonthMinimum: input.pricePerMonthMinimum,
+      pricePerMonthMaximum: input.pricePerMonthMaximum,
+      availablePaymentMethods: input.availablePaymentMethods
```

Comment:
Why do we define the available payment methods in the plan? and what if one of the payment plans after that got deactivated?

Reply:
because the users that want to become a member choose a member plan and in there we have the information which payment methods they can use.

Reply:
Will implement a delete protection later which prohibits delete if it's being used.